exercise/staff_views.py

```diff
@@ -325,27 +325,34 @@ class NextUnassessedSubmitterView(ExerciseBaseView, BaseRedirectView):
     access_mode = ACCESS.ASSISTANT
 
     def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
-        # Query submitters who have not been assessed yet.
+        # Query submissions which have not been assessed yet.
         submitter = None
-        submitters = (UserProfile.objects
-            .filter(submissions__exercise=self.exercise)
+
+        submissions = (Submission.objects
+            .filter(exercise=self.exercise)
             .annotate(
                 count_assessed=Count(
-                    'submissions__id',
-                    filter=(Q(submissions__grader__isnull=False)),
+                    'id',
+                    filter=(Q(grader__isnull=False)),
                 ),
             )
-            .filter(count_assessed=0)
-            .order_by('id'))
+            .filter(count_assessed=0))
+
+        # filter only the latest submissions
+        submissions = (submissions.annotate(latest=Max('submission_time'))
+           .filter(submission_time=F('latest'))
+           .order_by('latest'))
 
         previous_user_id = request.GET.get('prev')
         if previous_user_id:
-            # Find specifically the submitter AFTER the previously inspected one.
-            submitters_after = submitters.filter(id__gt=previous_user_id)
-            submitter = submitters_after.first()
+            previous_submission_time = submissions.filter(submitters__id=previous_user_id).first().submission_time
```

Comment:
When I was testing the code, this line produced an ``AttributeError: 'NoneType' object has no attribute 'submission_time'``. 

In my test, there was one submission by student1 and two submissions by student2 after the student1 submission. After manually assessing the student1 submission and clicking on the ``assess next submitter manually`` button the error occurred.

This error did not occur if there was two submissions by student1 before two submissions of student2. It only occurred if student1 had only one submission.

There bug should be fixed.

Reply:
Nice! I tested the changes that you made and the scenario I described seems to be handled correctly now.
I think this is ready to be merged.